action/actctx.go

```diff
@@ -40,6 +41,21 @@ func (act *AbstractAction) GasPrice() *big.Int {
 	return p.Set(act.gasPrice)
 }
 
+// IsExternal returns false for AbstractAction
+func (act *AbstractAction) IsExternal() bool {
```

Comment:
isExternal is a parameter at the Envelope level, AbstractAction does not need to know about this concept.

Also suggest change to isRLP rather than isExternal

Reply:
Envelope is the only level that needs to know this parameter

Reply:
yes, Envelope is the only level that needs to know this, but it is called upon Envelope's payload
```
func (elp *Envelope) Hash() hash.Hash256 {
	if elp.payload.IsExternal() {
		return elp.payload.RawHash()
	}
	return hash.Hash256b(elp.Serialize())
}
```
so this needs to be included in the `actionPayload` interface

if i remove this, the code won't compile

Reply:
in my first version, i did use `isRLP`
but later you call out `extern chain ID` for the new field's name in our discussion, then i think `isExternal` is more comprehensive? so we could potentially use this for other type of serialization, not particularly RLP
leave me another comment if you do want to use isRLP, i am fine with it